doc/man7/EVP_PKEY-RSA.pod

```diff
@@ -132,6 +132,13 @@ The RSA "e" value. The value may be any odd number greater than or equal to
 65537. The default value is 65537.
 For legacy reasons a value of 3 is currently accepted but is deprecated.
 
+=item "rsa-derive-from-pq"  (B<OSSL_PKEY_PARAM_RSA_DERIVE_FROM_PQ>) <unsigned integer>
+
+Indicate that our exponents and coefficients should be derived when only primes are 
+provided in the OSSL_PARAM array passed to EVP_PKEY_fromdata.  Requires at least
+that OSS_PKEY_PARAM_RSA_FACTOR1, OSS_PKEY_PARAM_RSA_FACTOR2, OSSL_PKEY_PARAM_RSA_E, 
+OSSL_PKEY_PARAM_RSA_N, and OSSL_PKEY_PARAM_RSA_D be provided
```

Comment:
We could also derive n (but it assumes n exists currently).

Reply:
Yeah, we could.  The current fromdata path assumes that n is provided via EVP_PKEY_fromdata, and errors if it isn't (prior to and with this changeset).  Think we should add derivation if its not provided?

Reply:
It seems like if we are going the route where we auto-derive parameters which *can* be derived, we should probably allow all theoretically derivable parameters to be derived for consistency.

I would maybe add a note to the manpage also that setting this does not cause re-derivation if the given parameters are already present.

Reply:
so whats the whole enumeration look like then:
1) If we have p and q, we can derive n, thats easy, but not yet handled
2) We can't derive e, as its selected, but we could choose the default of 65537, if its not provided, not yet handled in the fromdata path
3) We can derive d if its not provided, but e is given/selected, not yet handled
4) Our exponents/coefficients are already being computed, thats handled
5) If any of 1-4 are provided, we don't derive them, taking the provided values, potentially allowing EVP_PKEY_check to fail on incongruent values.

does that sound about right?

Reply:
(1) yeah, that would be fine to derive, although we always mandated it.
(2) no, missing e -> import should fail
(3) ok, although we always mandated it.
(5) ok

Reply:
Ok, I've updated the branch to derive 1,3, and 4 above if derive_from_pq is set.  E must be provided as per 2